### PR TITLE
implement first signature

### PR DESCRIPTION
**Pull request by itaysk**

### CODE REVIEW

File: tracee-rules/signatures/golang/anti_debugging_ptraceme.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+func (sig *antiDebuggingPtraceme) GetSelectedEvents() ([]types.SignatureEventSelector, error) {
+	return []types.SignatureEventSelector{{
+		Source: "tracee",
+		Name:   "ptrace",
+	}}, nil
+}
+
+func (sig *antiDebuggingPtraceme) OnEvent(e types.Event) error {
+	const PTRACE_TRACEME int64 = 0
```
Comment (yanivagman) [inline tracee-rules/signatures/golang/anti_debugging_ptraceme.go:37]: This is a bug.
Ptrace request are received as string.
Opened PR #387 to fix this

**Review by yanivagman — COMMENTED** [review]
There was a bug in ptrace request in tracee.
The signature will not work after the fix

File: tracee-rules/signatures/golang/anti_debugging_ptraceme.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	return nil
+}
+
+func (sig *antiDebuggingPtraceme) GetMetadata() (types.SignatureMetadata, error) {
+	return types.SignatureMetadata{
+		Name:        "detect self debugging using PTRACE_ME",
+		Description: "Process uses anti-debugging technique to block debugger",
+		Tags:        []string{"linux", "container"},
+		Properties: map[string]interface{}{
+			"Severity": 3,
```
[the remark this answers is not in the document]
Comment (itaysk) [reply tracee-rules/signatures/golang/anti_debugging_ptraceme.go:24]: I think we'll have a dedicated future PR that touches on the signature metadata

**Comment by itaysk** [comment]
cc: @roikol

File: tracee-rules/signatures/golang/anti_debugging_ptraceme.go (hunk 1 above)
Comment (yanivagman) [reply tracee-rules/signatures/golang/anti_debugging_ptraceme.go:37]: This will not work after rebasing on main

Comment (itaysk) [reply tracee-rules/signatures/golang/anti_debugging_ptraceme.go:37]: I don't understand, the type of ptrace request is not int64?

Comment (itaysk) [reply tracee-rules/signatures/golang/anti_debugging_ptraceme.go:37]: oh ok, you fixed it so it will print a string. got it

Comment (itaysk) [reply tracee-rules/signatures/golang/anti_debugging_ptraceme.go]: fixed

File: tracee-rules/signatures/golang/anti_debugging_ptraceme.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	if !ok {
+		return fmt.Errorf("invalid event")
+	}
+	if ee.EventName != "ptrace" {
+		return fmt.Errorf("invalid event")
+	}
+	if ee.ArgsNum > 0 && ee.Args[0].Name == "request" {
+		val, ok := ee.Args[0].Value.(string)
+		if !ok {
+			return fmt.Errorf("ptrace request arg must be long (int64)")
```
Comment (yanivagman) [inline tracee-rules/signatures/golang/anti_debugging_ptraceme.go:49]: Bad error message

Comment (itaysk) [reply tracee-rules/signatures/golang/anti_debugging_ptraceme.go]: I have replaced this code in the next commit.. but fixed anyway. thanks

File: tracee-rules/signatures/golang/signatures_test.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import "github.com/aquasecurity/tracee/tracee-rules/types"
+
+// sigTest is a utility to test signatures
+type sigTest struct {
+	// events are a series of events that the signature will be tested with (in the provided order)
+	events []types.Event
+	// provide a callback for this test case only if you want to check the Finding result
+	// if providing your own callback, make sure to call setStatus from it
+	cb types.SignatureHandler
+	// expect declares if we expect the signature to masth for this event
```
Comment (yanivagman) [inline tracee-rules/signatures/golang/signatures_test.go:12]: masth typo

**Review by yanivagman — APPROVED** [review]

File: tracee-rules/signatures/golang/signatures_test.go (hunk 4 above)
Comment (itaysk) [reply tracee-rules/signatures/golang/signatures_test.go:12]: 🤦‍♂️ sorry, thanks

**Pull request merged**